fitbenchmarking/core/fitting_benchmarking.py

```diff
@@ -410,12 +413,48 @@ def loop_over_jacobians(controller, options, grabbed_output):
                 controller.jacobian = jacobian
                 try:
                     with grabbed_output:
-                        # Calls timeit repeat with repeat = num_runs and
-                        # number = 1
-                        runtime_list = timeit.Timer(
-                            setup=controller.prepare,
-                            stmt=controller.fit
-                        ).repeat(num_runs, 1)
+
+                        def Timeit():
+                            """
+                            Function that performs fitting
+                            and times it, to be executed
+                            within a thread. Output is
+                            saved to temp file so it can
+                            be accessed outside of thread.
+                            """
+                            runtime = timeit.Timer(
+                                setup=controller.prepare,
+                                stmt=controller.fit
+                                ).repeat(num_runs, 1)
+                            temp_file = "timeit.pickle"
+                            with open(temp_file, 'wb') as f:
+                                dill.dump(runtime, f)
+
+                        set_runtime_error_flag = False
+
+                        # Set up thread to run Timeit function
+                        t = threading.Thread(target=Timeit)
+                        t.start()
+                        t.join(options.max_runtime)
+
+                        # if thread is still running after the
+                        # max runtime has elapsed, then stop
+                        # the thread and raise MaxRuntimeError
+                        if t.is_alive():
+                            t.join()
```

Comment:
Does this stop the thread? I thought join just waited patiently?

Reply:
Good point - there doesn't appear to be any way to stop threads, so I have switched to using processes, using the `multiprocess` package which also works on windows